src/src/com/microsoft/aad/adal/BasicWebViewClient.java

```diff
@@ -150,10 +149,23 @@ public void onPageFinished(WebView view, String url) {
 
     @Override
     public void onPageStarted(WebView view, String url, Bitmap favicon) {
-        Logger.v(TAG + "onPageStarted", "page is started with the url " + url);
+        final Uri uri = Uri.parse(url);
+        if (doesLoadingUrlContainAuthCode(url)) {
+            Logger.v(TAG, "Webview starts loading: " + uri.getHost() + uri.getPath() 
+                + " Auth code is returned for the loading url.");
+        } else {
+            Logger.v(TAG, "Webview starts loading: " + uri.getHost() + uri.getPath(), 
+                "Full loading url is: " + url, null);
+        }
+        
         super.onPageStarted(view, url, favicon);
         showSpinner(true);
     }
+    
+    private boolean doesLoadingUrlContainAuthCode(final String url) {
+        final Uri uri = Uri.parse(url);
```

Comment:
will throw if url is null. do we want to add a check and return false  in that case?